### DISCUSSION TITLE
Is it reasonable that time cost of diagonalization by scalapack_gvx is 7 times longer than that by genelpa

### DISCUSSION THREAD

**Discussion by pxlxingliang** [deepmodeling/abacus-develop#2295] [Q&A]
### Describe the bug

For the below example, the total time of the LCAO SCF calculation by scalapack_gvx is about 3 times longer than that by genelpa, and the longer of diagonalization (hamiltSolvePsiK) is 7 times.

The detail of the time cost is:
![image](https://user-images.githubusercontent.com/91927439/224954747-c144bd55-b732-4c8b-b1be-7646975523ba.png)
The left one is for scalapack_gvx  and the right one is for genelpa.

[006_Ti15.zip](https://github.com/deepmodeling/abacus-develop/files/10966598/006_Ti15.zip)
 

### To Reproduce

set `ks_solver` to be `scalapack_gvx` and `genelpa` respectively, and run the job with command `OMP_NUM_THREADS=1 mpirun -np 16 abacus`

### Environment

Bohrium image: registry.dp.tech/dptech/abacus:3.1.0

**Comment by Alcanderian** [comment]
Comparing with [this commit](https://github.com/deepmodeling/abacus-develop/commit/fb46766d878c4078301acfa1ff94b9e1f7252f75)
, I found test/performance/P105_si512_lcao takes 20 time slower in the lastest develop branch.

Perhaps the same reason for this issue?

**Comment by pxlxingliang** [comment]
> 

It seems not a same issue. In this issue the slower is compared to genelpa, and is found in version V3.1.0 (may also exist in earlier versions).
